pkg/patterns/addon/pkg/apis/v1alpha1/common_types.go

```diff
@@ -50,6 +50,9 @@ type CommonStatus struct {
 	Healthy bool     `json:"healthy"`
 	Errors  []string `json:"errors,omitempty"`
 	Phase   string   `json:"phase,omitempty"`
+	// Conditions follows the API specification "Conditions" properties.
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
So as this is, whenever controllers update and recompile they will immediately get these conditions added to their CRD.

If we did it automatically we would need a release note at the very least, and I'm not sure we should force this on people.  The other option is that we could make it more modular, and we could create something like Patches where users can mix-in the behaviours that they want.

(I'm also wondering if we should just use reflection, instead of requiring users to implement the interface.  I think if it's a well-known type it's not unreasonable, and it is more user-friendly.)

Reply:
I changed the condition to use the patch-like approach and verified in my own operator: 
```
--> `all manifests are ready` 
--> delete a Deployment 
--> condition shows `ManifestsNotReady` 
--> wait for deployment to auto recover 
--> condition shows `all manfiests are ready` again . 
```
The change is in the last commit 
[`f4fd903` (#308)](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/308/commits/f4fd903209085de8188d5799c2646a6cf779433f)